### DISCUSSION TITLE
UI and UX Design: Reasoning/Objections/Suggestions

### DISCUSSION THREAD

**Discussion by U-C-S** [U-C-S/Hurl#131] [General]
A space to 
- put my reasoning for a specific UI related changes
- put you users/your objections to a specific change that's already made

Once a thread is resolved I'll hide it.

**Comment by U-C-S** [comment]
![image](https://github.com/U-C-S/Hurl/assets/50218121/20629c4f-82cd-47af-af83-2c8586a485e5)

(Not released, but a [commit](https://github.com/U-C-S/Hurl/commit/13e3d4d79d65e388e952b570c91392ddc3c5720b) in main branch)

Hid the buttons for Settings, Ruleset Manager, Timed default under a drop-down button.

## Reasoning
- Want the selection window to be minimal
- Want to introduce further updates where Hurl Settings and Ruleset manager are directly available as start menu shortcuts

**Reply by U-C-S** [reply]
Please comment here, under this thread instead of commenting under conversation itself to discuss about this change.

**Reply by Jay-o-Way** [reply]
Looks nice, but creates a big negative impact in terms of accessibility. Try putting your mouse away and navigate the window, and use Screen Reader...

**Reply by U-C-S** [reply]
Tried that, It's accessible, just underneath a extra step.

At this point, if anyone wants to directly modify Rules or the upcoming Settings App, A solution like start menu shortcut and Tray context menu option are much easily accessible.

**Reply by Jay-o-Way** [reply]
This button cannot be used in this position, in combination with a `wpfui:TitleBar`. The TitleBar will block the button.

**Reply by U-C-S** [reply]
> The TitleBar will block the button.

but It doesn't

**Reply by Jay-o-Way** [reply]
idea: Top-Left looks nice too...
![image](https://github.com/user-attachments/assets/a6be57e6-b744-46c2-8e85-0c2b32c8a7b9)
Until the single-window-with-a-Navigation-menu solution

**Reply by U-C-S** [reply]
> Top-Left looks nice too

Yes, it's looking nice, but I want to implement single-window-with-a-Navigation-menu part next... For now, I will leave dropdown button as it is currently.

**Comment by Jay-o-Way** [comment]
### URL control: TextBox

Users want things to be as fast and easy as possible. This control has a lot of important benefits with accessibility: Tabstop, in-place-editing, built-in "clear" button and context-menu, screen reader support... People that use non-standard input devices really need this. "Hiding" this behind a click-event means low accessibility and more interaction, frustrating many people.

> Editing URL from within Hurl window which I believe is a rare interaction

This is very suggestive. Sometimes I click on a link and want to change it a little bit before opening it in the browser. Maybe to remove a language part, like ".../en/..." or simply to go to the homepage. Also, this gives a (manual) possibility to do https://github.com/U-C-S/Hurl/discussions/124. No need to make things _extra_ difficult.

Small, specific preferences like position and (background) color are no problem.

**Reply by Jay-o-Way** [reply]
@niels9001 @riotrah your thoughts?

**Reply by U-C-S** [reply]
For context of discussion, I want to hide this TextBox under a button. such that, Clicking the button would change the button itself into a Textbox.
